src/detect-engine-content-inspection.c

```diff
@@ -245,10 +250,11 @@ static int DetectEngineContentInspectionInternal(DetectEngineThreadCtx *det_ctx,
             }
 
             /* If the value came from a variable, make sure to adjust the depth so it's relative
-             * to the offset value.
+             * to the offset value. Make sure to include distance iff it's not already included
              */
-            if (cd->flags & (DETECT_CONTENT_DISTANCE_VAR|DETECT_CONTENT_OFFSET_VAR|DETECT_CONTENT_DEPTH_VAR)) {
-                 depth += offset;
+            if (cd->flags & ((distance_included ? 0 : DETECT_CONTENT_DISTANCE_VAR) |
```

Comment:
Is it the same as `if (!distance_included &&  (cd->flags & list of flags))` ?

Reply:
They're not the same ... `distance_included` exsists so it (distance) isn't included twice. The other `DETECT` flags remain relevant.